src/registerCommand.ts

```diff
@@ -14,9 +18,27 @@ const registerCommand = () =>
     // Updating the configuration will trigger the `onDidChangeConfiguration`
     // handler which will correctly update the text and icon in the status bar.
     affectsConfiguration.forEach(setting => {
-      configuration.update(
-        setting,
-        !isFormattingActivated,
+      if (!setting.startsWith(EDITOR_CODE_ACTIONS_ON_SAVE_PATH)) {
+        return configuration.update(
+          setting,
+          !isFormattingActivated,
+          ConfigurationTarget.Global,
+        )
+      }
+
+      const codeActionsOnSaveConfiguration = workspace.getConfiguration(
+        EDITOR_CODE_ACTIONS_ON_SAVE_PATH,
+      )
+      const codeActionsOnSaveSetting =
+        setting.split(EDITOR_CODE_ACTIONS_ON_SAVE_PATH + '.').at(-1) || ''
+      const newCodeActionsOnSaveConfiguration = {
+        ...codeActionsOnSaveConfiguration,
+        [codeActionsOnSaveSetting]: !isFormattingActivated,
+      }
```

Comment:
Damn. I would have expected `codeActionsOnSaveConfiguration.update(codeActionsOnSaveSetting, !isFormattingActivated, ConfigurationTarget.Global)` to work, but now I understand why you had to do the spread.